### DISCUSSION TITLE
Help on fixing the behaviour of TerminalView UI

### DISCUSSION THREAD

**Discussion by ghost** [migueldeicaza/SwiftTerm#313] [Q&A]
Hi,

while after pulling your last code have fixed the mouse pointer selection even after the main terminal page has scrolled down, I still have two strange behaviours:

- when i press button Up or Down to scroll the command history, eg. if I press Up, Down, Down and Up
I get this printed out: `^[[A^[[B^[[B^[[A`. If I press Enter, these are going to be executed. I'm pretty sure that before pulling your latest changes this wasn't happening but i'm not 100% sure here. 

- when i type something and press Enter, this is going to be printed at next line and then the output of the command is printed on the following lines. That's not what happens with other terminals.

Pretty sure I'm doing something wrong.

**Comment by ghost** [comment]
Hope this helps to understand the issue:

<img width="409" alt="image" src="https://github.com/migueldeicaza/SwiftTerm/assets/129745430/6730c253-f49e-4a9d-9279-e6437c69d520">

**Reply by ghost** [reply]
Any help?

**Comment by migueldeicaza** [comment]
Sorry, I am on vacation right now.

Is this happening with the Mac sample or yours?

**Reply by ghost** [reply]
Mine, how can I test your Mac sample? Which project are you referring at?

**Reply by migueldeicaza** [reply]
In the repository there is a “terminal apps” directory with two projects: one for iOS and one for Mac.

Test with the Mac one
